pkg/webhook/common.go

```diff
@@ -153,6 +156,17 @@ func getServerConfig(mgr manager.Manager) *webhook.Server {
 	server.TLSMinVersion = *tlsMinVersion
 	if *clientCAName != "" {
 		server.ClientCAName = *clientCAName
+		server.TLSOpts = []func(*tls.Config){
+			func(cfg *tls.Config) {
+				cfg.VerifyPeerCertificate = func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
```

Comment:
Use VerifyConnection instead and check the slice length before indexing into it.